src/lint_problems.ts

```diff
@@ -0,0 +1,279 @@
+    if (!workspacePath) {
+        logOutput("ERROR", 'Workspace folder not found.');
+        return actionResultFailure("Unable to get workspace path.");
+    }
+
+    try {
+        const absoluteFolderPath = path.join(workspacePath, relativeFolder);
+        const normalizedFolderPath = normalizePath(absoluteFolderPath);
+        if (!isPathNeuroSafe(normalizedFolderPath)) {
+            return actionResultNoAccess(normalizedFolderPath);
+        }
+        if (!fs.existsSync(normalizedFolderPath)) {
+            return actionResultFailure(`Folder '${relativeFolder}' does not exist.`);
+        }
+
+        const diagnostics = vscode.languages.getDiagnostics();
+        const folderDiagnostics = diagnostics.filter(([uri]) => {
+            return uri.fsPath.startsWith(normalizedFolderPath) && isPathNeuroSafe(uri.fsPath);
+        });
+
+        if (folderDiagnostics.length === 0) {
+            return actionResultNoAccess(normalizedFolderPath);
+        }
+
+        const formattedDiagnostics = folderDiagnostics.map(([uri, diags]) => {
+            const relative = path.relative(workspacePath, uri.fsPath);
+            return getFormattedDiagnosticsForFile(relative, diags);
+        }).join('\n');
+
+        NEURO.client?.sendContext(`Linting problems for folder ${relativeFolder}: ${formattedDiagnostics}`);
+    } catch (error) {
+        NEURO.client?.sendContext(`Failed to get linting diagnostics for folder '${relativeFolder}'.`);
+        logOutput("ERROR", `Getting diagnostics for folder ${relativeFolder} failed: ${error}`);
+    }
+    return actionResultAccept();
+}
+
+// Handle diagnostics for the entire workspace
+export function handleGetWorkspaceLintProblems(actionData: ActionData): ActionResult {
+    if (!hasPermissions(PERMISSIONS.accessLintingAnalysis)) {
+        return actionResultNoPermission(PERMISSIONS.accessLintingAnalysis);
+    }
+
+    const workspacePath = getWorkspacePath();
+    if (!workspacePath) {
+        logOutput("ERROR", 'Workspace folder not found.');
+        return actionResultFailure("Unable to get workspace path.");
+    }
+
+    try {
+        const diagnostics = vscode.languages.getDiagnostics();
+        const safeDiagnostics = diagnostics.filter(([uri]) => isPathNeuroSafe(uri.fsPath));
+
+        if (safeDiagnostics.length === 0) {
+            return actionResultNoAccess(workspacePath);
+        }
+
+        const formattedDiagnostics = safeDiagnostics.map(([uri, diags]) => {
+            const relative = path.relative(workspacePath, uri.fsPath);
+            return getFormattedDiagnosticsForFile(relative, diags);
+        }).join('\n\n');
+
+        NEURO.client?.sendContext(`Linting problems for the current workspace: ${formattedDiagnostics}`);
+    } catch (error) {
+        NEURO.client?.sendContext("Couldn't get diagnostics for the workspace.");
+        logOutput("ERROR", `Failed to get diagnostics for workspace. Error: ${error}`);
+    }
+
+    return actionResultAccept();
```

Comment:
There is no need for separate context as everything is synchronous
```suggestion
        return actionResultAccept(`Linting problems for the current workspace: ${formattedDiagnostics}`);
    } catch (error) {
        logOutput("ERROR", `Failed to get diagnostics for workspace. Error: ${error}`);
        return actionResultFailure("Couldn't get diagnostics for the workspace.");
    }
```